rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -1192,6 +1192,127 @@ void saveSvpFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSa
+
+            assertEquals(Optional.empty(), svpFundTransactionUnsignedHash);
+
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_whenThereIsNoProposedFederationSavedNorSet_shouldReturnEmpty() {
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
+            assertFalse(svpFundTransactionUnsignedHash.isPresent());
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_postLovell700WithHashSetButNotInStorage_shouldReturnTheHash() {
+
+            // Arrange
+
+            Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+            federationStorageProvider.setSvpFundTransactionUnsignedHash(expectedHash);
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
+
+            // Assert
+
+            assertTrue(svpFundTransactionUnsignedHash.isPresent());
+            assertEquals(expectedHash, svpFundTransactionUnsignedHash.get());
+
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_postLovell700WithHashSetAndSavedInStorage_shouldReturnTheHash() {
+
+            // Arrange
+
+            Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+            federationStorageProvider.setSvpFundTransactionUnsignedHash(expectedHash);
+            federationStorageProvider.save(networkParameters, activations);
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
+
+            // Assert
+
+            assertTrue(svpFundTransactionUnsignedHash.isPresent());
+            assertEquals(expectedHash, svpFundTransactionUnsignedHash.get());
+
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_postLovell700AndHashDirectlySavedInStorage_shouldReturnTheHash() {
+
+            // Arrange
+
+            Sha256Hash hash = BitcoinTestUtils.createHash(123_456_789);
+            storageAccessor.saveToRepository(SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils.serializeSha256Hash(hash));
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
+
+            // Assert
+
+            assertTrue(svpFundTransactionUnsignedHash.isPresent());
+            assertEquals(hash, svpFundTransactionUnsignedHash.get());
+
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_postLovell700AndSetToNull_shouldReturnEmpty() {
+
+            // Arrange
+
+            federationStorageProvider.setSvpFundTransactionUnsignedHash(null);
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
```

Comment:
I think we should also test the case when a `null` value is in storage

Reply:
Updated.